front-end/src/lib/chartUtils.js

```diff
@@ -28,8 +26,9 @@ export const getSeriesData = async (response, filename, columns, minMax, chartTy
 
     // If not colour, return values in array to allow for boost
     if (chartType !== "colour") {
+        const timestampOffset = columns[headerIndices.x].header === 'Timestamp (ms)' ? new Date(columns[headerIndices.x].timespan.start + 'Z').getTime() - parseFloat(lines[0][headerIndices.x]): 0
         return lines.map((line) => {
-            return [parseFloat(line[headerIndices.x]), parseFloat(line[headerIndices.y])];
+            return [parseFloat(line[headerIndices.x]) + timestampOffset, parseFloat(line[headerIndices.y])];
```

Comment:
timestampOffset shouldn't cause us to run through all the data as well, please include this in the parsing directly so it only happens once :)